lib/SILOptimizer/ARC/ARCRegionState.cpp

```diff
@@ -531,7 +531,9 @@ void ARCRegionState::summarizeBlock(SILBasicBlock *BB) {
   SummarizedInterestingInsts.clear();
 
   for (auto &I : *BB)
-    if (!canNeverUseValues(&I) || I.mayReleaseOrReadRefCount() ||
+    // FIXME: mayReleaseOrReadRefCount should be a strict subset of
+    // canUseObject. If not, there is a bug in canUseObject.
+    if (canUseObject(&I) || I.mayReleaseOrReadRefCount() ||
```

Comment:
If it should be a strict subset, why not just put in an assert (maybe you do not feel comfortable with relying since you haven't done a go over?).

Even if you don't feel comfortable with that (i.e. you want us to be conservatively correct in non-asserts builds), we should have an assert here to track down the issues.

Reply:
The code is currently dead and the FIXME is just speculation. I added it as a note to whomever looks at the code again soon in case we temporarily resurrect it. I think that will be the time to clean it up. I didn't want to potentially change the original behavior without testing it. At some not-to-distant point the code will be deleted anyway.